src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -192,31 +199,55 @@ public void BuildSymbolDefinitionMap(ISymbolNode[] definedSymbols)
             }
         }
 
-        public void EmitSymbolDefinition(int currentOffset, NodeFactory factory)
+        private string GetSymbolToEmitForTargetPlatform(string symbol)
+        {
+            string symbolToEmit = symbol;
+
+            if (_targetPlatform.OperatingSystem == TargetOS.OSX)
+            {
+                // On OSX, we need to prefix an extra underscore to account for correct linkage of 
+                // extern "C" functions.
+                symbolToEmit = "_"+symbol;
```

Comment:
The bootstrapper is small part of the problem. It is needed to link with PInvokes or  RuntimeImports.